src/plugins/formulas/__tests__/formulas.spec.js

```diff
@@ -1595,6 +1663,36 @@ describe('Formulas general', () => {
         ['y'], ['=A3'], ['=A6'], ['=A5'], ['=A8'], ['=A7'], ['=A10'], ['=A9'], ['=A12']
       ]);
     });
+
+    it('autofill suports validation', async() => { 
+      const hot = handsontable({ 
+        data: [ 
+          [1, 2, 3],
+          ['txt', 1, 2], 
+    ], 
+      formulas: { 
+        engine: HyperFormula
+       }, 
+      columns: { 
+        type: 'numeric' 
+      }, 
+    }); 
+
+      selectCell(0, 0); 
+      autofill(0, 2); 
+
+      expect(hot.getSourceData()).toEqual([[1, 1, 1], ['txt', 1, 2]]); 
+      expect($(getCell(1, 0)).hasClass(hot.getSettings().invalidCellClassName)).toBe(false); 
+      expect($(getCell(2, 0)).hasClass(hot.getSettings().invalidCellClassName)).toBe(false); 
+
+      selectCell(1, 0); 
+      autofill(1, 2); 
+
+      await sleep(100);
+      expect(hot.getSourceData()).toEqual([[1, 1, 1], ['txt', 'txt', 'txt']]); 
+      //expect($(getCell(1, 1)).hasClass(hot.getSettings().invalidCellClassName)).toBe(true); 
```

Comment:
@wszymanski I think this test should pass, but it doesn't. Is it not a bug?

Reply:
Currently, the tests contain the options schema that is not supported by Hot (follow the [comment](https://github.com/handsontable/handsontable/pull/8092/files#r638488649)). If I guess the intention with column settings the cell validity state should be set as `false` for the three cells.

Reply:
As @budnix said, test provided by you doesn't have valid configuration. I will revert change please commit your changes within separate PR 🙂